File: web/src/services/notification.ts
import request from '../utils/request'
import type { PageResult } from './audit'

export interface NotificationItem {
  id: number
  user_id: number
  title: string
  content: string
  type: string  // info / success / warning / error
  source: string
  source_id: number
  read: boolean
  created_at: string
}

export function getNotifications(params?: {
  read?: string
  page?: number
  page_size?: number
}): Promise<PageResult<NotificationItem>> {
  return request.get('/notifications', { params })
}

export function getUnreadCount(): Promise<{ count: number }> {
  return request.get('/notifications/unread-count')
}

export function markRead(id: number): Promise<void> {
  return request.put(`/notifications/${id}/read`)
}

export function markAllRead(): Promise<void> {
  return request.put('/notifications/read-all')
}

export function deleteNotification(id: number): Promise<void> {
  return request.delete(`/notifications/${id}`)
}

File: internal/dto/notification.go
package dto

// NotificationResponse 通知响应
type NotificationResponse struct {
	ID        uint   `json:"id"`
	UserID    uint   `json:"user_id"`
	Title     string `json:"title"`
	Content   string `json:"content"`
	Type      string `json:"type"`
	Source    string `json:"source"`
	SourceID  uint   `json:"source_id"`
	Read      bool   `json:"read"`
	CreatedAt string `json:"created_at"`
}

// NotificationUnreadCount 未读数响应
type NotificationUnreadCount struct {
	Count int64 `json:"count"`
}
